TechnicalDocuments/SnowplowHowTo.md

```diff
@@ -0,0 +1,163 @@
+# Posting Analytics Events to Snowplow
+
+Snowplow is a data centric analytics platform that combines flexibility with reasonable type safety. This is achieved by verifying incoming events against a user defined schema before storing them. The same schema definitions are used to define database tables for storing analytics events. That events are stored in well defined database tables makes it possible to match front end analytics with backend log data. The downside is that becomes slightly more complicated to post front end analytics events.
+
+Snowplow has published a schema with two frequently used event types. The first is for screen views and the other is based on the fields in a Google Analytics event. In Snowplow lingo they are referred to as screen view event (`SPScreenView`) and structured events (`SPStructered`). Events with a bespoke definition are referred as unstructured events (`SPUnstructured`)
+
+# Posting Screen View Events from the iOS Client.
+
+View controllers that are created with `BentoKit` and use the Babylon specialisation of `BoxViewController` will post screen view events if the view model conforms to `ScreenNaming`.
+
+```swift
+final class AwesomeRenderer: BoxRenderer {
+  ...
+}
+
+extension HomeRenderer: ScreenNaming {
+     static var AwesomeRenderer: String { return ScreenNames.BestFeatureEver.awesome }
+}
+```
+
+Screen names should be inside the `ScreenNames` namespace to avoid name clashes. Declaring the view model to conform to `ScreenNaming` will also assign the screen name as the default accessibility identifier.
+
+View controllers that are not based on `BentoKit` should post a `ScreenEvent` in `viewWillAppear`.
+
+```swift
+class MassiveViewController: UIViewController {
+...
+    override func viewWillAppear(_ animated: Bool) {
+       super.viewWillAppear(animated)
+
+        Current.analyticsService.track(ScreenEvent.viewWillAppear(screen: ScreenNames.OldSchoolFeature.massive))
```

Comment:
I understand the exercise being done here but I still think it would be better to have a ViewModel and forward the `viewWillAppear` call to it and trigger the relevant event there over having this directly in the ViewController as we don't use MVC.

Reply:
Working with legacy code it is always a judgment call wether to do the minimal change or make an improvement to the modelling. I would not create a `ViewModel`, but if there is a legacy one I think it is okay to add a `viewWillAppear` method it.